openshift/scripts/generate-dockerfiles.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+#
+# This script generates the productized Dockerfiles
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+function install_generate_hack_tool() {
+  local tmp_dir
+  tmp_dir="$(mktemp -d)"
+
+  git clone https://github.com/openshift-knative/hack.git "${tmp_dir}"
```

Comment:
Is this clone necessary? Wouldn't `go install github.com/openshift-knative/hack/cmd/generate@latest` suffice? And if we do need clone could we limit clone depth via `--depth` flag?

Reply:
```
go install github.com/openshift-knative/hack/cmd/generate@latest
go: downloading github.com/openshift-knative/hack v0.0.0-20240905080328-cc95ad425341
go: github.com/openshift-knative/hack/cmd/generate@latest (in github.com/openshift-knative/hack@v0.0.0-20240905080328-cc95ad425341):
	The go.mod file for the module providing named packages contains one or
	more replace directives. It must not contain directives that would cause
	it to be interpreted differently than if it were the main module.
```

But good point to limit the depth of the clone

Reply:
updated

Reply:
Ahh the replace thingy...

Reply:
> Ahh the replace thingy...
yep :/ (had the same issue in other repos as well :/ )